### DISCUSSION TITLE
Driver for Murata 2AE/CYW4373

### DISCUSSION THREAD

**Discussion by victorallume** [micropython/micropython#14048] [Libraries & Drivers]
I'm trying to get the Murata 2AE/CYW4373 rolling with micropython. I gather the first thing I need to do is to get the firmware converted to a format for https://github.com/georgerobotics/cyw43-driver/tree/main/firmware - but I'm not certain where the firmware files are initially obtained from - I've found https://github.com/murata-wireless/cyw-fmac-fw/  and https://github.com/Infineon/wifi-host-driver/tree/master/WiFi_Host_Driver/resources/firmware
(The former seems more likely as it has the clm_blob). I'm also unsure where the version number for the `combined.h` comes from e.g. `w43439A0_7_95_49_00_combined.h` - where does 7_95_49_00 come from? 

Anything else I'm likely to need/traps for young players?

**Comment by dpgeorge** [comment]
The `murata-wireless/cyw-fmac-fw` repository should be the correct one.

The version number can be found by inspecting the binary firmware blob, it should be at the end of the blob.  Use eg `xxd` to dump the binary and see the version string.

> Anything else I'm likely to need/traps for young players?

I suspect there will need to be some minor updates to the driver code, but to find that out you'd just need to try it, ie try the updated firmware blobs with the new hardware.

**Reply by victorallume** [reply]
Thanks. I'm also seeing there's an nvram file - I've taken the one from https://github.com/murata-wireless/cyw-fmac-nvram/blob/b5a9984658e16db6bd50ba6b585231dc6760e1eb/cyfmac4373-sdio.2AE.txt and converted to null-seperated strings as per the other nvram files - is this the right location to get it from?

**Reply by dpgeorge** [reply]
Yes, you definitely need the correct nvram data.

That one you link to looks correct, for the 2AE module.
